pkg/beacon/relay/groupselection/groupselection.go

```diff
@@ -16,6 +17,14 @@ import (
 
 var logger = log.Logger("keep-groupselection")
 
+const (
+	// Duration of one ticket submission round in blocks.
+	ticketSubmissionRoundDuration = 3
+
+	// Duration of the mining lag in blocks.
+	miningLag = 6
+)
```

Comment:
Since `chainConfig.TicketSubmissionTimeout` is supplied as a parameter from the outside, we should add some sanity check:
```
// Recommended parameters all clients should use to minimize their expenses.
// It is not a must to obey but it is nice and polite. And being nice to others
// helps in reducing own costs because other clients should respect the same
// protocol.
const (
	// the number of blocks one round takes
	roundDuration = uint64(3)
	// the delay in blocks after all rounds complete to ensure all transactions
	// are mined
	miningLag     = uint64(6)
)

// calculateRoundsCount takes the on-chain ticket submission timeout
// and calculates the number of rounds for ticket submission. If it is not 
// possible to use the recommended round duration and mining lag because the 
// supplied timeout is too short, function returns an error.
func calculateRoundsCount(submissionTimeout uint64) (uint64, error) {
	if submissionTimeout-miningLag <= roundDuration {
		return 0, fmt.Errorf("submission timeout is too short")
	}

	return (submissionTimeout - miningLag) / roundDuration, nil		
}
```

Reply:
Not worth doing it now but I'll drop the idea in the comment in case we'll be interested in longer round durations in the future.
```
type roundParameters struct {
	// the total number of rounds
	roundCount uint64
	// the number of blocks one round takes
	roundDuration uint64
	// the delay in blocks after all rounds complete to ensure all transactions
	// are mined
	miningLag uint64
}

// Recommended parameters all clients should use to minimize their expenses.
// It is not a must to obey but it is nice and polite. And being nice to others
// helps in reducing own costs because other clients should respect the same
// protocol.
const (
	defaultRoundDuration = uint64(3)
	defaultMiningLag     = uint64(6)
)

// calculateRoundCount takes the on-chain ticket submission timeout
// and calculates the expected round parameters. If it is not possible to use
// 
func calculateRoundParams(submissionTimeout uint64) (*roundParameters, error) {
	if submissionTimeout-defaultMiningLag <= defaultRoundDuration {
		return nil, fmt.Errorf("submission timeout is too short")
	}

	return &roundParameters{
		roundCount:    (submissionTimeout - defaultMiningLag) / defaultRoundDuration,
		roundDuration: defaultRoundDuration,
		miningLag:     defaultMiningLag,
	}, nil
}
```

Reply:
Good point! Changed this as you proposed.